### PR TITLE
Add feature request for copy initializers with 'ref' intent arguments

### PR DESCRIPTION
**Pull request by bradcray**
This adds a test that demonstrates that we get an unhelpful error message
when defining copy initializers that take their argument by `ref` intent and
asks the question whether we should add support for such copy initializers.
See copyWithRefOther.future for details.

This PR also renames the previous `copyWithRefOther.chpl` to
`copyWithConstRefOther.chpl` since that's a more accurate description of
what it was testing.

### CODE REVIEW

**Comment by bradcray** [comment]
@mppf / @lydia-duncan : This follows up on the email I sent last night / this morning.  I'll file a future after merging.

File: test/types/records/init/copy/copyWithRefOther.chpl [hunk 1]
```diff
@@ -1,15 +1,20 @@
 record R {
   var x: int;
-
+
```
Comment (lydia-duncan) [inline test/types/records/init/copy/copyWithRefOther.chpl]: Looks like you added a tab accidentally here

**Comment by lydia-duncan** [comment]
Seems reasonable to me!

File: test/types/records/init/copy/copyWithRefOther.chpl (hunk 1 above)
Comment (bradcray) [reply test/types/records/init/copy/copyWithRefOther.chpl]: Yep!

**Pull request merged**